core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/DynamicPartitionFileSetParquetSink.java

```diff
@@ -0,0 +1,202 @@
+import co.cask.cdap.etl.api.batch.BatchSink;
+import co.cask.cdap.etl.api.batch.BatchSinkContext;
+import co.cask.hydrator.common.HiveSchemaConverter;
+import co.cask.hydrator.plugin.common.StructuredToAvroTransformer;
+import org.apache.avro.generic.GenericRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import parquet.avro.AvroParquetInputFormat;
+import parquet.avro.AvroParquetOutputFormat;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * A {@link BatchSink} to write Parquet records to a {@link PartitionedFileSet}.
+ */
+@Plugin(type = "batchsink")
+@Name("DynamicPFSParquet")
+@Description("Sink for a PartitionedFileSet that writes data in Parquet format and uses a dynamic partition key.")
+public class DynamicPartitionFileSetParquetSink extends
+  PartitionedFileSetSink<Void, GenericRecord> {
+  public static final String NAME = "DynamicPFSParquet";
+  public static final String PARTITION_COL_PREFIX = "p_";
+
+  private static final Logger LOG = LoggerFactory.getLogger(DynamicPartitionFileSetParquetSink.class);
+  private static final String SCHEMA_DESC = "The Parquet schema of the record being written to the Sink as a JSON " +
+    "Object.";
+  private static final String FIELD_DESC = "The fields to be used for the partitions as comma separated values.";
+
+  private StructuredToAvroTransformer recordTransformer;
+  private final DynamicPartitionParquetSinkConfig config;
+
+  public DynamicPartitionFileSetParquetSink(DynamicPartitionParquetSinkConfig config) {
+    super(config);
+    this.config = config;
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext context) {
+    Map<String, String> sinkArgs = getAdditionalPFSArguments();
+    PartitionedFileSetArguments.setDynamicPartitioner(sinkArgs, FieldValueDynamicPartitioner.class);
+    context.addOutput(Output.ofDataset(config.name, sinkArgs));
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    super.configurePipeline(pipelineConfigurer);
+    String pfsName = config.name;
+    String basePath = config.basePath == null ? pfsName : config.basePath;
+
+    String schema = config.schema.toLowerCase();
+    // parse to make sure it's valid
+    new org.apache.avro.Schema.Parser().parse(schema);
+    String hiveSchema;
+    Schema parsedSchema;
+    try {
+      parsedSchema = Schema.parseJson(schema);
+      hiveSchema = HiveSchemaConverter.toHiveSchema(parsedSchema);
+    } catch (UnsupportedTypeException | IOException e) {
+      throw new RuntimeException("Error: Schema is not valid ", e);
+    }
+
+    Partitioning.Builder partitionBuilder = Partitioning.builder();
+    String[] partitionFields = config.fieldNames.split(",");
+    for (int i = 0; i < partitionFields.length; i++) {
+      // We need to make the name of the partitioned column unique to avoid 'Column repeated in partitioning columns'
```

Comment:
Oh; nevermind, I think I understand now. The partitioningFields would have the same name as the columns in the schema?